app/models/concerns/account/counters.rb

```diff
@@ -79,6 +50,28 @@ def account_stat
 
   private
 
+  def updated_account_stat(key, value)
+    AccountStat.upsert(
+      initial_values(key, value),
+      on_duplicate: duplicate_values(key, value),
+      unique_by: :account_id
+    )
+  end
+
+  def initial_values(key, value)
+    { :account_id => id, key => [value, 0].max }.tap do |values|
+      values.merge!(last_status_at: Time.current) if key == :statuses_count
+    end
+  end
+
+  def duplicate_values(key, value)
+    Arel.sql(
+      ["#{key} = (account_stats.#{key} + #{value})", 'updated_at = CURRENT_TIMESTAMP'].tap do |values|
+        values << 'last_status_at = CURRENT_TIMESTAMP' if key == :statuses_count && value.positive?
+      end.join(', ')
+    )
+  end
```

Comment:
I wonder how we could make those less awkward and more readable… I'm honestly not sure.